myapp/spec/system/users_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :system do 
+  describe 'signup page' do 
+    it 'should correctly render components' do 
+      visit users_signup_path
+
+      expect(page).to have_content('User Signup')
+      expect(page).to have_button('Back to login')
+      expect(page).to have_field('Username')
+      expect(page).to have_field('Password')
+      expect(page).to have_field('Password confirmation')
+      expect(page).to have_button('Sign Up')
+    end
+
+    context 'when input invalid value' do 
+      it 'should show error message for username less than 3 chars' do 
+        visit users_signup_path 
+
+        fill_in 'Password', with: '12345'
+        fill_in 'Password confirmation', with: '12345'
+        # find("input[type='password']").set('12345')
+        
+        click_on 'Sign Up'
+        
+        expect(page).to have_content('Username should be between 3 and 20 chars!')
```

Comment:
Would be better to use I18n because we have both lang (en & ja) and check both message.

Reply:
Applied I18n in 61bcb23